app/controllers/provider_interface/application_choices_controller.rb

```diff
@@ -11,5 +14,15 @@ def show
       @application_choice = GetApplicationChoicesForProviders.call(providers: current_provider_user.providers)
         .find(params[:application_choice_id])
     end
+
+  private
+
+    def query_mapping
```

Comment:
Nice solution to the injection problem! This might be more readable as a pure function, eg `query_mapping(sort_by, sort_order)`. Out of interest, what's the behaviour if you pass garbage as the `sort_order` param?

Reply:
👍